index.js

```diff
@@ -66,12 +58,18 @@ module.exports = (opts) => {
         if (req.body) {
           if (req.body instanceof Stream) {
             body = req.body
-          } else if (typeof req.body === 'string') {
-            body = req.body
-            populateHeaders(headers, body, 'text/plain')
           } else {
-            body = JSON.stringify(req.body)
-            populateHeaders(headers, body, 'application/json')
+            // Per RFC 7231 §3.1.1.5 if this header is not present we MAY assume application/octet-stream
+            const contentType = req.headers['content-type'] || 'application/octet-stream'
+            // detect if body should be encoded as JSON
+            // supporting extended content-type header formats:
+            // - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type
+            const shouldEncodeJSON = contentType.toLowerCase().indexOf('application/json') === 0
+            // transparently support JSON encoding
+            body = shouldEncodeJSON ? JSON.stringify(req.body) : req.body
```

Comment:
I am fine with this. However it creates a breaking change, before we used `populateHeaders(headers, body, 'application/json')`
Let's keep it in mind for the release notes and create a major version.

Reply:
Yeah.. I was bending my mind a little bit around this one. The tests in this repo passes with the above change. What will this exactly break (so I can understand what it going on here better)?

Anyhow, a major version of this alignment are kinda given I think.

Reply:
Before, we were assuming the `application/json` contentType for objects in req.body. It might be that some users are relying on this convenient JSON support. 

A major version here would be fine. Thanks.